src/vendor/azure-uamqp-c/src/messaging.c

```diff
@@ -188,11 +188,20 @@ AMQP_VALUE messaging_delivery_rejected(const char* error_condition, const char*
                 }
                 else
                 {
-                    if (rejected_set_error(rejected, error_handle) != 0)
+                    if((error_info != NULL) &&
+                    (error_set_info(error_handle, error_info) != 0))
                     {
-                        LogError("Cannot set error on REJECTED state handle");
+                        LogError("Cannot set error info on error AMQP value for REJECTED state");
                         error_constructing = true;
                     }
+                    else
+                    {
+                        if (rejected_set_error(rejected, error_handle) != 0)
+                        {
+                            LogError("Cannot set error on REJECTED state handle");
+                            error_constructing = true;
+                        }
+                    }
                 }
 
                 error_destroy(error_handle);
```

Comment:
Do we need to destroy the fields as well?
We should confirm whether `error_set_info` does an `amqpvalue_clone` - and if it does we will need to destroy this copy of it I think.

Reply:
I don't think we need to destroy `fields` here, `error_set_info` clean up the resources it creates.
https://github.com/Azure/azure-uamqp-python/blob/bd30f8f3b6a8f5708858409367ef335ec8ffce1e/src/vendor/azure-uamqp-c/src/amqp_definitions.c#L571-L603

and it's an object being passed from the python layer, so python would garbage collect that.

(And I checked `AMQP_VALUE messaging_delivery_modified(bool delivery_failed, bool undeliverable_here, fields message_annotations)`, it doesn't destroy message_annotations)